xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-ui/src/main/resources/XWiki/Like/UserProfileUIX.xml

```diff
@@ -254,30 +165,53 @@
     </property>
     <property>
       <content>{{velocity}}
-=$escapetool.xml($services.localization.render("like.userprofile.menu"))=
-#set($columns = ["doc.title", "doc.location", "like"])
+=$escapetool.xml($services.localization.render('like.userprofile.menu'))=
 #set ($docTitleDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.title')))
 #set ($docLocationDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.location')))
 #set ($likesDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.likes')))
-#set($columnsProperties = {
-  "doc.title" : { "displayName": $docTitleDisplayName, "type": "text", "size": 30, "link": "view", "sortable": false, "filterable": false },
-  "doc.location" : { "displayName": $docLocationDisplayName, "type": "text", "size": 30, "html": true, "sortable": false, "filterable": false },
-  "like" : { "displayName": $likesDisplayName, "type" : "number", "sortable": false, "filterable": false }
-})
-
-#set ($currentUser = $services.model.serialize($doc.documentReference, "default"))
-#set ($pageToCallReference = $services.model.resolveDocument("XWiki.Like.UserProfileUIX"))
-#set ($pageToCall = $xwiki.getDocument($pageToCallReference))
-#set ($queryParams = {
-  "user": $currentUser,
-  "livetable": "true",
-  "xpage": "plain",
-  "outputSyntax": "plain"
-})
-#set($options = {
-  'url': $pageToCall.getURL('get', $escapetool.url($queryParams))
+#set ($currentUser = $services.model.serialize($doc.documentReference, 'default'))
+#set ($liveDataConfig = {
+  'meta': {
+    'propertyDescriptors': [
+       {
+         'id': 'doc.title',
+         'displayer': 'link',
+         'sortable': false,
+         'filterable': false,
+         'name': $docTitleDisplayName
+       },
+       {
+         'id': 'doc.location',
+         'displayer': 'html',
+         'sortable': false,
+         'filterable': false,
+         'name': $docLocationDisplayName
+       },
+       {
+         'id': 'like',
+         'displayer': 'number',
+         'sortable': false,
+         'filterable': false,
```

Comment:
same

Reply:
shouldn't the types be autodiscovered?

Reply:
yes sure, my comment was not about the like but more generic. I put it after the column declarations, my bad it wasn't clear. In any case you explained it to me and I saw the code re the defaults. Thx

Reply:
> Now for the like column we could imagine that a default would be defined by the source (with a future LD source). It would allow to specify it in one place and have all LD using it inherit from it (unless they really need to change the type but it's a rare case IMO).

Yes, that's the plan.